src/Extensions/Abstractions/ReplayableLogs/TimedScopeDefinition.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+
+namespace Microsoft.Omex.Extensions.Abstractions
+{
+	/// <summary>
+	/// Store Timed Scope name
+	/// </summary>
+	public readonly struct TimedScopeDefinition
+	{
+		/// <summary>
+		/// Timed Scope name
+		/// </summary>
+		public string Name { get; }
+
+
+		/// <summary>
+		/// TimedScopeDefinition constructor
+		/// </summary>
+		/// <param name="name">Timed Scope name</param>
+		/// <exception cref="ArgumentException">Thrown when name null or empty</exception>
+		public TimedScopeDefinition(string name) =>
+			Name = string.IsNullOrWhiteSpace(name)
+				? throw new ArgumentException("TimedScope name must not be empty or null") // Activity won't be stated in case of an empty string
+				: name;
+
+
+		/// <summary>
+		/// Indicates whether the current object is equal to another object of the same type
+		/// </summary>
+		/// <param name="other">An object to compare with this object</param>
```

Comment:
Missing `<returns>`. Applies throughout this class.

Reply:
https://github.com/microsoft/Omex/pull/153#discussion_r391957268